#!/usr/bin/env python
PACKAGE = "peripheralsInterface"

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("ballhandlerPlotEnabled", bool_t, 0, "Enable Data Dump for plotting", False)
gen.add("Kp", int_t, 0, "BallHandler P-value of PID", 1500, 0, 65536)
gen.add("Ki", int_t, 0, "BallHandler I-value of PID", 0, 0, 65536)
gen.add("Kd", int_t, 0, "BallHandler D-value of PID", 0, 0, 65536)
gen.add("iTh", int_t, 0, "Threshold of using I (iMin)", 0, 0, 65536)
gen.add("iMax", int_t, 0, "Limits the maximum I value", 4000, 0, 100000)

gen.add("Ang_Kp", int_t, 0, "BallHandler Angle P-value of PID", 8000, 0, 65536)
gen.add("Ang_Ki", int_t, 0, "BallHandler Angle I-value of PID", 0, 0, 65536)
gen.add("Ang_Kd", int_t, 0, "BallHandler Angle D-value of PID", 0, 0, 65536)

gen.add("Angle", int_t, 0, "BallHandler Angle", 50, 0, 100) 
gen.add("BallPossession", int_t, 0, "BallHandler BallPossession Threshold", 20, 0, 100) 

gen.add("PwmMax", int_t, 0, "BallHandler PWM Maximum power (max=648)", 648, 0, 648)
gen.add("PwmMaxDeltaSize", int_t, 0, "BallHandler PWM maximum step size per cycle (2.5ms, 400Hz, max=648<<16)", 65536, 0, 10*65536)

gen.add("Feedforward_vel_factor", double_t, 0, "BallHandler Feedforward Factor", 0.1, 0.0, 1.0)

gen.add("MaxAngleLeftArm", int_t, 0, "Max BallHandler Angle for Left BH arm", 4000, 0, 65536) 
gen.add("MaxAngleRightArm", int_t, 0, "Max BallHandler Angle for Right BH arm", 4000, 0, 65536) 

exit(gen.generate(PACKAGE, "BallHandlers", "ballhandlers"))
